### PR TITLE
Added a check when storing an existing preset, now requires you to up…

### LINKED ISSUE #139: presets: it should trigger an error when trying to create a preset that already exists

### PR DESCRIPTION
**Pull request by saidketchman**
…date and will not store it

Resolves #139

### CODE REVIEW

File: db/memory.go [hunk 1]
```diff
@@ -43,6 +43,10 @@ func (r *memoryDatabase) StorePreset(preset types.Preset) (types.Preset, error)
 	r.mtx.Lock()
 	defer r.mtx.Unlock()
 
+	if _, ok := r.presets[preset.Name]; ok {
```
Comment (saidketchman) [inline db/memory.go]: I wanted to call RetrivePreset here too but the locks stopped me. Didn't come up with a way to resolve it so went this way for now.

File: db/db_test.go [hunk 2]
```diff
@@ -59,6 +60,14 @@ var _ = Describe("Database", func() {
 				Expect(err).NotTo(HaveOccurred())
 				Expect(res).To(Equal(preset))
 			})
+
+			It("should fail to store a preset if it already exists", func() {
+				dbInstance.StorePreset(preset)
+				res, err := dbInstance.StorePreset(preset)
+				Expect(res).To(Equal(types.Preset{}))
```
Comment (saidketchman) [inline db/db_test.go]: Should I just ignore the res in this case since I care just about the error?

Comment (flavioribeiro) [inline db/db_test.go]: yep I think you can just ignore it

**Comment by flavioribeiro** [comment]
LGTM. Thanks @saidketchman! 👍

**Pull request merged**